subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/XCTestIntegrationTest.groovy

```diff
@@ -0,0 +1,226 @@
+
+        when:
+        file("src").deleteDir()
+        newTestApp.writeToProject(testDirectory)
+        succeeds("test")
+
+        then:
+        executedAndNotSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        newTestApp.expectedSummaryOutputPattern.matcher(output).find()
+    }
+
+    def "execute added test suite and case"() {
+        def oldTestApp = new SwiftXcTestTestApp([
+            newTestSuite("FooTestSuite", [
+                newTestCase("testA", TestElement.TestCase.Result.PASS)
+            ])
+        ])
+
+        def newTestApp = new SwiftXcTestTestApp([
+            newTestSuite("FooTestSuite", [
+                newTestCase("testA", TestElement.TestCase.Result.PASS),
+                newTestCase("testB", TestElement.TestCase.Result.PASS)
+            ]),
+            newTestSuite("BarTestSuite", [
+                newTestCase("testA", TestElement.TestCase.Result.PASS)
+            ])
+        ])
+
+        given:
+        oldTestApp.writeToProject(testDirectory)
+
+        when:
+        succeeds("test")
+
+        then:
+        executedAndNotSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        oldTestApp.expectedSummaryOutputPattern.matcher(output).find()
+
+        when:
+        file("src").deleteDir()
+        newTestApp.writeToProject(testDirectory)
+        succeeds("test")
+
+        then:
+        executedAndNotSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+        newTestApp.expectedSummaryOutputPattern.matcher(output).find()
+    }
+
+    def "test tasks are up-to-date when nothing changes between invocation"() {
+        def testApp = new SwiftXcTestTestApp([
+            newTestSuite("PassingTestSuite", [
+                newTestCase("testPass", TestElement.TestCase.Result.PASS)
+            ])
+        ])
+
+        given:
+        testApp.writeToProject(testDirectory)
+
+        when:
+        succeeds("test")
+        succeeds("test")
+
+        then:
+        skipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
+    }
+
+    def "xctest component can specify a dependency on another library"() {
+        def lib = new SwiftLib()
+        def testApp = new SwiftXcTestTestApp([
+            newTestSuite("PassingTestSuite", [
+                newTestCase("testPass", TestElement.TestCase.Result.PASS, "XCTAssert(sum(a: 40, b: 2) == 42)")
+            ], ["Greeter"])
+        ])
+
+        given:
+        settingsFile << """
+include 'Greeter'
+"""
+        buildFile << """
+project(':Greeter') {
+    apply plugin: 'swift-library'
+}
+
+dependencies {
+    testImplementation project(':Greeter')
+}
+"""
+        lib.writeToProject(file('Greeter'))
+        testApp.writeToProject(testDirectory)
+
+        when:
+        succeeds("test")
+
+        then:
+        executedAndNotSkipped(":compileTestSwift", ":linkTest", ":createXcTestBundle", ":xcTest", ":test")
```

Comment:
I will add the assertion. The test as it is uses the `sum` function from the `Greeter` module.